parse/NamedValueRefParser.cpp

```diff
@@ -0,0 +1,131 @@
+#include "Parse.h"
+
+#include "ParseImpl.h"
+#include "ConditionParser.h"
+#include "ValueRefParser.h"
+
+#include "MovableEnvelope.h"
+#include "../universe/ValueRefs.h"
+#include "../universe/ValueRef.h"
+#include "../util/Directories.h"
+
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/phoenix_core.hpp>
+
+#include <typeinfo>
+
+#define DEBUG_PARSERS 0
+
+#if DEBUG_PARSERS
+namespace std {
+    inline ostream& operator<<(ostream& os, const std::map<int, int>&) { return os; }
+    inline ostream& operator<<(ostream& os, const std::map<std::string, std::map<int, int>>&) { return os; }
+}
+#endif
+
+namespace parse {
+
+    template <typename T>
+    void insert_named_ref(std::map<std::string, std::unique_ptr<ValueRef::AnyValueRef>>& named_refs,
+                          const std::string& name,
+                          const parse::detail::MovableEnvelope<ValueRef::ValueRef<T>>& ref_envelope,
+                          bool& pass)
+    {
+        ErrorLogger() << "Registering from named_values.focs.txt : " << name << " ! ValueRef<" << typeid(T).name() << ">";
+        if (true) {
+         ErrorLogger() << "Registering from named_values.focs.txt : opening envelope";
+         std::unique_ptr<ValueRef::ValueRef<T>> ref = std::move(ref_envelope.OpenEnvelope(pass));
+         ErrorLogger() << "Registering from named_values.focs.txt : add a AnyValueRef to named_refs map";
+         named_refs.emplace(name, std::move(std::unique_ptr<ValueRef::AnyValueRef>(std::move(ref))));
```

Comment:
can probably just move from `ref` to pass... I suspect it would implicitly cast? If not, could perhaps `static_pointer_cast` to the base class?

If not, don't need to move from the newly-constructed `unique_ptr<AnyValueRef>` as it should already be an rvalue reference.

Reply:
not sure why that didnt work before. Shrunk it down to the most simple variant.